stunnel/stunnel_cache.mli

```diff
@@ -39,3 +39,5 @@ val flush : unit -> unit
 
 (** GCs old stunnels *)
 val gc : unit -> unit
+
+val set_legacy_protocol_and_ciphersuites_allowed : bool -> unit
```

Comment:
Discussed whether this could be the only way of setting this value to avoid any possible race conditions and then we can hide the `bool ref` in the Stunnel module. Discussed the lock free approach to handling this `ref` and it seems to be safe as is so long as we don't go and add any other direct setters using the ref directly.